### PR TITLE
Updated command `slcli image detail`

### PR DESCRIPTION
**Pull request by edsonarios**
Issue: #1860
```
softlayer-python/ (issue1860) $ slcli image detail 5706
┌────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│               name │ value                                                                                                                                              │
├────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│                 id │ 5706                                                                                                                                            │
│  global_identifier │ d6447b06-bae6-4a40                                                                                                               │
│               name │ Backup template for disk migration of 'testttbootmode.softlayer.ibm.com'.                                                                          │
│             status │ Active                                                                                                                                             │
│ active_transaction │                                                                                                                                                    │
│            account │ 30                                                                                                                                             │
│            created │ 2020-08-07T12:12:14-07:00                                                                                                                          │
│         total_size │ 5.46G                                                                                                                                              │
│         visibility │ Private                                                                                                                                            │
│               type │ System                                                                                                                                             │
│               flex │ False                                                                                                                                              │
│               note │ Note │
│        datacenters │ ┌───────┬───────┬─────────────┐                                                                                                                    │
│                    │ │  DC   │ size  │ transaction │                                                                                                                    │
│                    │ ├───────┼───────┼─────────────┤                                                                                                                    │
│                    │ │ par01 │ 5.46G │      -      │                                                                                                                    │
│                    │ └───────┴───────┴─────────────┘                                                                                                                    │
│      virtual disks │ ┌───────────────────────────────┬──────────┬──────────────┐                                                                                        │
│                    │ │            Device             │ capacity │ size on disk │                                                                                        │
│                    │ ├───────────────────────────────┼──────────┼──────────────┤                                                                                        │
│                    │ │ CentOS 7.0-64  │  25 GB   │    5.46G     │                                                                                        │
│                    │ │        106-SWAP         │   2 GB   │     N/A      │                                                                                        │
│                    │ └───────────────────────────────┴──────────┴──────────────┘                                                                                        │
│        share image │ ┌─────────┬───────────────────────────┐                                                                                                            │
│                    │ │ Account │         shared on         │                                                                                                            │
│                    │ ├─────────┼───────────────────────────┤                                                                                                            │
│                    │ │  468  │ 2023-03-01T09:14:38-06:00 │                                                                                                            │
│                    │ │ 305  │ 2023-03-01T07:23:32-06:00 │                                                                                                            │
│                    │ │ 123  │ 2023-03-01T11:27:17-06:00 │                                                                                                            │
│                    │ └─────────┴───────────────────────────┘                                                                                                            │
└────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

### CODE REVIEW

File: SoftLayer/CLI/image/detail.py [hunk 1]
```diff
@@ -85,3 +87,51 @@ def _get_transaction_groups(children_images):
         transactions.add(utils.lookup(child, 'transaction', 'transactionGroup', 'name'))
 
     return transactions
+
+
+def _get_virtual_disks(children_images):
+    """Returns image details as datacenter, size, and transaction within a formatting table.
```
Comment (BrianSantivanez) [inline SoftLayer/CLI/image/detail.py:93]: Change this comment line accord the method task

File: SoftLayer/CLI/image/detail.py [hunk 2]
```diff
@@ -85,3 +87,51 @@ def _get_transaction_groups(children_images):
...
+                device_name,
+                capacity,
+                size_on_disk
+            ])
+
+    return table_virtual_disks
+
+
+def _get_share_image(image):
+    """Returns image details as datacenter, size, and transaction within a formatting table.
```
[the remark this answers is not in the document]
Comment (edsonarios) [reply SoftLayer/CLI/image/detail.py:124]: Done

File: SoftLayer/CLI/image/detail.py [hunk 3]
```diff
@@ -52,9 +54,9 @@ def cli(env, identifier):
                    )])
     table.add_row(['flex', image.get('flexImageFlag')])
     table.add_row(['note', image.get('note')])
-    table.add_row(['created', image.get('createDate')])
-    table.add_row(['total_size', formatting.b_to_gb(total_size)])
     table.add_row(['datacenters', _get_datacenter_table(children_images)])
+    table.add_row(['virtual disks', _get_virtual_disks(children_images)])
+    table.add_row(['share image', _get_share_image(image)])
```
Comment (caberos) [inline SoftLayer/CLI/image/detail.py:60]: please a capital letter the name column

File: SoftLayer/CLI/image/detail.py [hunk 4]
```diff
@@ -85,3 +87,51 @@ def _get_transaction_groups(children_images):
...
     return transactions
+
+
+def _get_virtual_disks(children_images):
+    """Returns datacenters as a table.
+
+      :param children_images: A list of images.
+      """
+
+    table_virtual_disks = formatting.Table(['Device', 'capacity', 'size on disk'])
```
Comment (caberos) [inline SoftLayer/CLI/image/detail.py:98]:
```suggestion
    table_virtual_disks = formatting.Table(['Device', 'Capacity', 'Size on disk'])
```

File: SoftLayer/CLI/image/detail.py [hunk 5]
```diff
@@ -85,3 +87,51 @@ def _get_transaction_groups(children_images):
...
+
+    return table_virtual_disks
+
+
+def _get_share_image(image):
+    """Returns shares image as a table.
+
+      :param image: Detail information about image.
+      """
+    table_share_account = formatting.Table(['Account', 'shared on'])
```
Comment (caberos) [inline SoftLayer/CLI/image/detail.py:128]:
```suggestion
    table_share_account = formatting.Table(['Account', 'Shared on'])
```

**Review by caberos — CHANGES_REQUESTED** [review]

File: SoftLayer/CLI/image/detail.py [hunk 6]
```diff
@@ -85,3 +87,51 @@ def _get_transaction_groups(children_images):
...
+
+def _get_virtual_disks(children_images):
+    """Returns datacenters as a table.
+
+      :param children_images: A list of images.
+      """
+
+    table_virtual_disks = formatting.Table(['Device', 'capacity', 'size on disk'])
+
+    if utils.lookup(children_images[0], 'blockDevices'):
```
Comment (allmightyspiff) [inline SoftLayer/CLI/image/detail.py:100]: Add a check to make sure children_images actually has things in it.

**Review by allmightyspiff — CHANGES_REQUESTED** [review]

File: SoftLayer/CLI/image/detail.py (hunk 3 above)
Comment (edsonarios) [reply SoftLayer/CLI/image/detail.py:60]: Done

File: SoftLayer/CLI/image/detail.py (hunk 5 above)
Comment (edsonarios) [reply SoftLayer/CLI/image/detail.py:128]: Done

File: SoftLayer/CLI/image/detail.py (hunk 6 above)
Comment (edsonarios) [reply SoftLayer/CLI/image/detail.py:100]: The function `lookup` check if the attribute exist

**Review by allmightyspiff — APPROVED** [review]

**Review by caberos — APPROVED** [review]
LGTM

**Pull request merged**